### DISCUSSION TITLE
How to specify endpoint description and name?

### DISCUSSION THREAD

**Discussion by ChillFish8** [poem-web/poem#142] [Q&A] [answered]
Hi! Loving poem and the open api integration but one thing that's been plaguing me is that redoc gets populated with *most* info but doesn't populate the endpoint name.

I also couldn't find a way to manually add this (`#[oai]`) came to mind but my vague attempts didn't work (although as future improvement it would be awsome if you could specify addition openapi attributes via the function's macro.

![image](https://user-images.githubusercontent.com/57491488/147617183-d560b388-9553-4571-896a-b5feff7d31b8.png)

example code:
```rust
pub struct UsersApi;

#[OpenApi]
impl UsersApi {
    #[oai(path = "/@me", method = "get")]
    pub async fn user(
        &self,
        session: Data<&Session>,
        token: TokenBearer,
    ) -> Result<Json<UserPayload>> {
        todo!()
    }
}```

**Comment by sunli829** [comment]
Do it like this:

```rust
pub struct UsersApi;

#[OpenApi]
impl UsersApi {
    /// description
    #[oai(path = "/@me", method = "get")]
    pub async fn user(
        &self,
        session: Data<&Session>,
        token: TokenBearer,
    ) -> Result<Json<UserPayload>> {
        todo!()
    }
}
```

**Reply by ChillFish8** [reply]
\>-< Kind of embarrassed that I never thought about doing that.

**Reply by ChillFish8** [reply]
Okay this works for the summary which is awesome! But how can you add a description if the rust side's description is produced as the title?

**Comment by sunli829** [comment, marked as answer]
The first line is the summary, and the following is the description. 😁

```rust
pub struct UsersApi;

#[OpenApi]
impl UsersApi {
    /// summary
    ///
    /// description
    /// A
    /// B
    #[oai(path = "/@me", method = "get")]
    pub async fn user(
        &self,
        session: Data<&Session>,
        token: TokenBearer,
    ) -> Result<Json<UserPayload>> {
        todo!()
    }
}
```

**Reply by ChillFish8** [reply]
Oh perfect! Thank you so much!
